packages/teams-js/src/internal/utils.ts

```diff
@@ -518,3 +518,49 @@ const supportsPerformanceTimers = !!performance && 'now' in performance;
 export function getCurrentTimestamp(): number | undefined {
   return supportsPerformanceTimers ? performance.now() + performance.timeOrigin : undefined;
 }
+
+/**
+ * @hidden
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * Function to check whether the data is a primitive type or a plain object.
+ * @param value The value to check
+ * @returns true if the value is a primitive type or a plain object, false otherwise
+ *
+ */
+export function isPrimitiveOrPlainObject(value: unknown, depth: number = 0): boolean {
+  if (depth > 1000) {
+    return false; // Limit recursion depth
+  }
+
+  // Check if the value is a primitive type or null
+  if (
```

Comment:
nit: what about this approach:
 ```function isPrimitiveOrNull(value) {
  return value === null || (typeof value !== 'object' && typeof value !== 'function');
}```
 it covers all primitive types without needing to list each one individually.

Reply:
The function needs to allow the nested primitives in an object as well. This check will incorrectly return false if the value was a plain object or an array of primitives.